src/neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -324,6 +324,19 @@ public virtual VerifyResult Verify(StoreView snapshot, BigInteger totalSenderFee
             long net_fee = NetworkFee - size * NativeContract.Policy.GetFeePerByte(snapshot);
             if (net_fee < 0) return VerifyResult.InsufficientFunds;
             if (!this.VerifyWitnesses(snapshot, net_fee)) return VerifyResult.Invalid;
+
+            if (Version == TransactionVersion.OracleResponse)
+            {
+                // Oracle response only can be signed by oracle nodes
+
+                var hashes = GetScriptHashesForVerifying(snapshot);
+
+                if (hashes.Length != 1 || hashes[0] != NativeContract.Oracle.GetOracleMultiSigAddress(snapshot))
+                {
```

Comment:
Maybe we don't need this check? As we already check it in `OracleContract.SetOracleResponse`.

```cs
[ContractMethod(0_03000000, ContractParameterType.Boolean, ParameterTypes = new[] { ContractParameterType.ByteArray, ContractParameterType.ByteArray }, ParameterNames = new[] { "transactionHash", "oracleResponse" })]
        private StackItem SetOracleResponse(ApplicationEngine engine, Array args)
        {
            if (args.Count != 2) return false;
            UInt160 account = GetOracleMultiSigAddress(engine.Snapshot);
            if (!InteropService.Runtime.CheckWitnessInternal(engine, account)) return false;
```

Reply:
And the more `TransactionVersion` we have, the more check conditions we'll have here.

Reply:
> Maybe we don't need this check? As we already check it in OracleContract.SetOracleResponse.
`OracleContract.SetOracleResponse` it's executed during `OnPersist`, this check it's for avoid people to use `OracleResponse` type

Reply:
You're right.